R/step-subset-filter.R

```diff
@@ -32,6 +32,24 @@ filter.dtplyr_step <- function(.data, ..., .preserve = FALSE) {
   step_subset_i(.data, i)
 }
 
+filter_by_lgl_col <- function(dots) {
+  if (length(dots) > 1) {
+    return(FALSE)
+  }
+
+  dot <- dots[[1]]
+
+  if (is_symbol(dot)) {
+    return(TRUE)
+  }
+
+  # catch expressions of form `!x`
+  is_call(dot) &&
```

Comment:
You can simplify this a bit by using more arguments to `is_call()`

Reply:
neat, actually this makes so much sense that I could have expected that from an `rlang` function!